### ISSUE TITLE
Enable topology features for CAPV

### ISSUE THREAD

**Issue by erkanerol** [giantswarm/roadmap#2340]

**Comment by erkanerol** [comment]
We will follow ***Spread Nodes Across Multiple Hosts in a Single Compute Cluster*** part of the doc https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.6/vmware-tanzu-kubernetes-grid-16/GUID-tanzu-k8s-clusters-multi-az-vsphere.html#spread-nodes-across-multiple-hosts-in-a-single-compute-cluster-1

I tried it. It failed because of wrong tags on the objects. I tagged `Cluster` as `k8s-region` and VMs as `k8s-zones`. Now, it fails because of a lack of VM Groups. The API requires both host groups and VM groups.

**Comment by erkanerol** [comment]
Update: Manuel asked IONOS to create VMGroups and we are waiting.

**Comment by erkanerol** [comment]
Update: We have problems regarding permissions to add VMs to VMGroups. Related thread https://gigantic.slack.com/archives/C01BYMF6RN0/p1684137963460649

**Comment by gawertm** [comment]
waiting for IONOS to give us the permissions to edit DRS rules / Add VMs to VM Groups. then we can go ahead testing this further. most likely @jkremser needs to take this over while Erkan is afk.
@erkanerol is there anything we need to know to take this over during your afk?

**Comment by erkanerol** [comment]
- Please add parameters to `nodeClasses` section in cluster-vsphere. [See](https://github.com/giantswarm/cluster-vsphere/blob/main/helm/cluster-vsphere/values.yaml#L126)
- Please ensure that changes in parameters trigger updates. If necessary, implement versioning in names as we do for machine templates. [See](https://github.com/giantswarm/cluster-vsphere/blob/main/helm/cluster-vsphere/templates/vspheremachinetemplate.yaml#L14)
- Follow "Spread Nodes Across Multiple Hosts in a Single Compute Cluster" in [the doc](https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.6/vmware-tanzu-kubernetes-grid-16/GUID-tanzu-k8s-clusters-multi-az-vsphere.html#spread-nodes-across-multiple-hosts-in-a-single-compute-cluster-1).
- In helm chart, render VSphereFailureDomain and VSphereDeploymentZone per nodeClass. Use common parameters from root values like datacenter, datastore etc.

**Comment by jkremser** [comment]
wip pr
https://github.com/giantswarm/cluster-vsphere/pull/68

**Comment by gawertm** [comment]
this one is blocked by IONOS permissions. need to test on other provider (neoedge)

**Comment by erkanerol** [comment]
@gawertm will we use neoedge for vsphere too?

**Comment by erkanerol** [comment]
Related: https://gigantic.slack.com/archives/C05G8A1F25D/p1691070600128209 CC: @vxav

**Comment by vxav** [comment]
Sorry @gawertm I missed that message somehow.

For NeoEdge we started with VCD only because the first quote for the vSphere environment included a 3 years commitment period so they were looking at alternative options. I enquired for an update 👇 
https://gigantic.slack.com/archives/C05L3JURA7P/p1692598602137839

**Comment by erkanerol** [comment]
@gawertm It doesn't make sense to keep this in "Blocked/Waiting" column. Let's move it to the backlog with low priority.
